zeppelin-web/src/app/credential/credential.controller.js

```diff
@@ -15,30 +15,135 @@
 'use strict';
 
 angular.module('zeppelinWebApp').controller('CredentialCtrl', function($scope, $route, $routeParams, $location, $rootScope,
-                                                                         $http, baseUrlSrv) {
+                                                                         $http, baseUrlSrv, ngToast) {
   $scope._ = _;
 
+  $scope.credentialInfo = [];
   $scope.credentialEntity = '';
   $scope.credentialUsername = '';
   $scope.credentialPassword = '';
-  
-  $scope.updateCredentials = function() {
+
+  $scope.showAddNewCredentialInfo = false;
+
+  var getCredentialInfo = function() {
+    $http.get(baseUrlSrv.getRestApiBase()+'/credential').
+      success(function(data, status, headers, config) {
+        $scope.credentialInfo = data.body;
+        console.log('Success %o %o', status, data.message);
+      }).
+      error(function(data, status, headers, config) {
+        console.log('Error %o %o', status, data.message);
+      });
+  };
+
+  $scope.addNewCredentialInfo = function() {
     $http.put(baseUrlSrv.getRestApiBase() + '/credential',
-      { 'entity': $scope.credentialEntity,
-        'username': $scope.credentialUsername,
-        'password': $scope.credentialPassword
-      } ).
+    {
+      'entity': $scope.credentialEntity,
+      'username': $scope.credentialUsername,
+      'password': $scope.credentialPassword
+    }).
     success(function (data, status, headers, config) {
-      alert('Successfully saved credentials');
-      $scope.credentialEntity = '';
-      $scope.credentialUsername = '';
-      $scope.credentialPassword = '';
+      BootstrapDialog.alert({
+        closable: true,
+        message: 'Successfully saved credentials.'
+      });
+      $scope.resetCredentialInfo();
+      getCredentialInfo();
+      $scope.showAddNewCredentialInfo = false;
       console.log('Success %o %o', status, data.message);
     }).
     error(function (data, status, headers, config) {
-      alert('Error saving credentials');
+      BootstrapDialog.alert({
+        closable: true,
+        message: 'Error saving credentials'
+      });
       console.log('Error %o %o', status, data.message);
     });
   };
 
+  $scope.cancelCredentialInfo = function() {
+    $scope.showAddNewCredentialInfo = false;
+    $scope.resetCredentialInfo();
+  };
+
+  $scope.resetCredentialInfo = function() {
+    $scope.credentialEntity = '';
+    $scope.credentialUsername = '';
+    $scope.credentialPassword = '';
+  };
+
+  $scope.copyOriginCredentialsInfo = function(entity, username, password) {
```

Comment:
It looks to me that this function doesn't do anything at all. it copies input arguments but doesn't store these copies anywhere or return anything